hawc/apps/assessment/forms.py

```diff
@@ -667,3 +667,66 @@ class Meta:
         model = models.Dataset
         fields = ("name", "description", "published")
         field_classes = {"description": QuillField}
+
+
+class AssessmentRootModelChoiceField(forms.ModelChoiceField):
+    def __init__(self, model, *args, **kwargs):
+        self.model = model
+        super().__init__(None, *args, **kwargs)
+
+    def set_choices(self, node, assessment):
+        root = self.model.get_assessment_root(assessment.pk)
+        queryset = self.model.get_tree(root)
+        if node.pk is not None:
+            queryset = queryset.exclude(path__startswith=node.path, depth__gte=node.depth)
+        self._queryset = self.model.annotate_nested_names(queryset)
+        if not self._queryset:
+            self.choices = []
+        else:
+            default_choice = (self._queryset[0].pk, "-----")
+            self.choices = [
+                default_choice,
+                *[(obj.pk, obj.nested_name) for obj in self._queryset[1:]],
+            ]
+
+
+class TagForm(forms.ModelForm):
+    # TODO make htmx not allow multiple open updates,
```

Comment:
make this a TODO for future work? Would this happen frequently, it seems like a possible rare edge case